llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -4699,7 +4696,8 @@ static SDValue lowerShuffleViaVRegSplitting(ShuffleVectorSDNode *SVN,
       // less an implementation question, and more a profitability one.
       return SDValue();
 
-    OutMasks[DstVecIdx].second.resize(ElemsPerVReg);
+    if (OutMasks[DstVecIdx].second.empty())
```

Comment:
I think you could use `resize(ElemsPerVReg, -1)`. The second argument is only used if the size increases.